js/ui/codeView.js

```diff
@@ -1201,6 +1217,24 @@ var CodeViewManager = new Lang.Class({
         this._sessions.splice(idx, 1);
     },
 
+    handleMinimizeWindow: function(actor) {
+        if (this._stopped)
+            return;
+
+        // First, determine if this is an app window getting destroyed
+        let session = this._getSession(actor, SessionLookupFlags.SESSION_LOOKUP_APP);
+        let eventType = SessionDestroyEvent.SESSION_DESTROY_APP_DESTROYED;
+
+        if (session)
+            this._removeSession(session, eventType);
+        else {
+            // if we're minimizing as toolbox, we switch back
```

Comment:
This is not really the desired behavior though...

The primary goal of the FtH implementation is to maintain an illusion that the front and back of an app/toolbox pair in a session really function as a single entity.
So, when the app is minimized, if there's a toolbox in the session, it should minimize together with it; when the app is unminimized, the toolbox should also unminimize with it, and viceversa. Again, the effect we want to create is that those are really the same physical window.

That's what the `applyWindowMinimizationState()` functions are trying to achieve here: https://github.com/endlessm/gnome-shell/blob/master/js/ui/codeView.js#L892-L923

Reply:
I did this workaround because in other case, we can't remove the CodingSession on minimize status, if we don't flip back, the app will be fliped when unminimize and the CodingSession has been removed.

So I think that if we don't flip back we can't remove the CodingSession on minimize because that will break the link and destroy the toolbox.

I'll try to handle this in the handleMapWindow, and avoid the toolbox destroy on minimize, to avoid the flip back.

Reply:
Sounds good. Yes, if the app was flipped when it was minimized (i.e. it was actually the toolbox to be minimized) upon unminimization should still be flipped, and viceversa.
Basically minimization should not have any impact on the coding sessions, if at all possible.